typescript/src/commands/Starwars/swfilm.ts

```diff
@@ -0,0 +1,118 @@
+})
+export class UserPaginatedMessageCommand extends PaginatedMessageCommand {
+	public async run(message: GuildMessage, args: PaginatedMessageCommand.Args) {
+		const { t } = args;
+		const [film, loadingMessage] = await Promise.all([args.rest('string'), sendLoadingMessage(message, t)]);
+
+		const results = await this.fetchAPI(film.toLowerCase());
+
+		if (results.length === 0) {
+			this.error(LanguageKeys.Commands.StarWars.FilmQueryFail, { film });
+		}
+
+		const display = new SkyraPaginatedMessage({
+			template: new MessageEmbed() //
+				.setColor(await this.context.db.fetchColor(message))
+				.setThumbnail(CdnUrls.StarWarsLogo)
+		});
+
+		const filmTitles = t(LanguageKeys.Commands.StarWars.FilmEmbedTitles);
+
+		for (const result of results) {
+			display
+				.addPageEmbed((embed) => {
+					const description = [
+						`**${filmTitles.episodeId}**: ${result.episodeId}`, //
+						`**${filmTitles.releaseDate}**: ${t(LanguageKeys.Globals.DateValue, { value: new Date(result.releaseDate) })}`, //
+						`**${filmTitles.director}**: ${result.director}`, //
+						`**${filmTitles.producers}**: ${t(LanguageKeys.Globals.AndListValue, { value: result.producers })}` //
+					].join('\n');
+
+					return embed
+						.setTitle(result.title) //
+						.setDescription(description);
+				})
+				.addPageEmbed((embed) =>
+					embed
+						.setTitle(result.title) //
+						.setDescription(cutText(result.openingCrawl, 4096))
+				)
+				.addPageEmbed((embed) => {
+					const description: string[] = [];
+
+					if (result.characters.length) {
+						description.push(
+							`**${filmTitles.characters}**: ${t(LanguageKeys.Globals.AndListValue, {
+								value: result.characters.map((character) => `\`${character.name}\``)
+							})}`
+						);
+					}
+
+					if (result.planets.length) {
+						description.push(
+							`**${filmTitles.planets}**: ${t(LanguageKeys.Globals.AndListValue, {
+								value: result.planets.map((planet) => `\`${planet.name}\``)
+							})}`
+						);
+					}
+
+					if (result.species.length) {
+						description.push(
+							`**${filmTitles.species}**: ${t(LanguageKeys.Globals.AndListValue, {
+								value: result.species.map((species) => `\`${species.name}\``)
+							})}`
+						);
+					}
+
+					if (result.starships.length) {
+						description.push(
+							`**${filmTitles.starships}**: ${t(LanguageKeys.Globals.AndListValue, {
+								value: result.starships.map((starship) => `\`${starship.name}\``)
+							})}`
+						);
+					}
+
+					if (result.vehicles.length) {
+						description.push(
+							`**${filmTitles.vehicles}**: ${t(LanguageKeys.Globals.AndListValue, {
+								value: result.vehicles.map((vehicle) => `\`${vehicle.name}\``)
+							})}`
+						);
+					}
+
+					return embed
+						.setTitle(result.title) //
+						.setDescription(description.join('\n\n'));
+				});
+		}
+
+		await display.run(loadingMessage, message.author);
+		return loadingMessage;
+	}
+
+	private async fetchAPI(film: string) {
```

Comment:
not sure if it's possible- but if so, might be a good idea to make a `StarWarsPaginatedCommand` class, and have the `fetchAPI` as a generic base method on the class, and inherit from that? `fetchAPI` seems to be pretty much duplicated with different type methods everywhere. If not possible then all good, though.

Reply:
Not worth it because of variables tbh